### PR TITLE
azurerm_nginx_deployment - allow in-place SKU change

### PR DESCRIPTION
**Pull request by arpith-f5**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description
NGINXaaS now supports in-place migration of SKU. The SKU change is only supported from standard_Monthly to standardv2_Monthly. Any other change in SKU will be rejected by the resource provider.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

Manually tested migrating SKU from standard_Monthly to standardv2_Monthly

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_nginx_deployment` - support SKU change

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/nginx/nginx_deployment_resource.go [hunk 1]
```diff
@@ -849,3 +850,21 @@ func expandConfiguration(model Configuration) nginxconfiguration.NginxConfigurat
...
+
+func (m DeploymentResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			rd := metadata.ResourceDiff
+			oldSKU, newSKU := rd.GetChange("sku")
+			oldSKUStr := oldSKU.(string)
+			newSKUStr := newSKU.(string)
+
+			if oldSKUStr != "standard_Monthly" || newSKUStr != "standardv2_Monthly" {
```
[the remark this answers is not in the document]
Comment (arpith-f5) [reply internal/services/nginx/nginx_deployment_resource.go]: Discussed with @puneetsarna on slack. We are in agreement on not adding validations to the provider, NGINX RP will reject unsupported migrations and the customer can delete and recreate their deployment.

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @arpith-f5. Is it possible to modify the existing `update` test so that we can validate and track that this remains updateable?

This property in the docs will also need updating.

**Comment by arpith-f5** [comment]
> Thanks for this PR @arpith-f5. Is it possible to modify the existing `update` test so that we can validate and track that this remains updateable?
> 
> This property in the docs will also need updating.

I don't think we should add an acceptance test for this. The only SKU migration we are supporting is from standard to standardv2. The standard plan is scheduled for retirement in the coming months so if I add a test that creates a deployment with standard plan(to update to StandardV2) then the test will start failing as you won't be able to create new deployments on the standard plan. This functionality is mainly for existing NGINXaaS deployments.

Updated the docs. Thanks!

**Review by stephybun — APPROVED** [review]
Thanks @arpith-f5, based on your explanation it makes sense to not add this to our tests. LGTM 👍

**Pull request merged**